pkg/autodiscovery/providers/file.go

```diff
@@ -199,54 +197,52 @@ func (c *FileConfigProvider) collectEntry(file os.FileInfo, path string, checkNa
 func (c *FileConfigProvider) collectDir(parentPath string, folder os.FileInfo) configPkg {
 	configs := []integration.Config{}
 	defaultConfigs := []integration.Config{}
-	metricConfigs := []integration.Config{}
+	otherConfigs := []integration.Config{}
 	const dirExt string = ".d"
 	dirPath := filepath.Join(parentPath, folder.Name())
 
 	if filepath.Ext(folder.Name()) != dirExt {
-		// the name of this directory isn't in the form `checkname.d`, skip it
+		// the name of this directory isn't in the form `integrationName.d`, skip it
 		log.Debugf("Not a config folder, skipping directory: %s", dirPath)
-		return configPkg{configs, defaultConfigs, metricConfigs}
+		return configPkg{configs, defaultConfigs, otherConfigs}
 	}
 
 	// search for yaml files within this directory
 	subEntries, err := ioutil.ReadDir(dirPath)
 	if err != nil {
 		log.Warnf("Skipping config directory %s: %s", dirPath, err)
-		return configPkg{configs, defaultConfigs, metricConfigs}
+		return configPkg{configs, defaultConfigs, otherConfigs}
 	}
 
 	// strip the trailing `.d`
-	checkName := strings.TrimSuffix(folder.Name(), dirExt)
+	integrationName := strings.TrimSuffix(folder.Name(), dirExt)
 
 	// try to load any config file in it
 	for _, sEntry := range subEntries {
 		if !sEntry.IsDir() {
 
-			entry := c.collectEntry(sEntry, dirPath, checkName)
+			entry := c.collectEntry(sEntry, dirPath, integrationName)
 			if entry.err != nil {
 				// logging already done in collectEntry
 				continue
 			}
 			// determine if a check has to be run by default by
-			// searching for check.yaml.default files
+			// searching for integration.yaml.default files
 			if entry.isDefault {
 				defaultConfigs = append(defaultConfigs, entry.conf)
-			} else if entry.isMetric {
-				metricConfigs = append(metricConfigs, entry.conf)
-			} else if entry.isLogsOnly {
-				// skip logs-only configs for now as they are not processed by autodiscovery
+			} else if entry.isMetric || entry.isLogsOnly {
+				otherConfigs = append(otherConfigs, entry.conf)
```

Comment:
Probably a bit of a detail, but still worth asking IMO: here I think handling "logs-only" integration configs like "metrics" integration configs has consequences on how the precedence of `.yaml` over `.yaml.default`  files is handled.

Could you clarify how this `.yaml` vs `.yaml.default` precedence should be handled when dealing with `logs-only` config. Is there a test that covers this?

(in general I'm thinking being consistent with "check" integration config would make sense)

Reply:
I think we should be fine thanks to this: https://github.com/DataDog/datadog-agent/pull/1724/files/659c0f9e2117748dfad07be4ee8f3bc9a47cdf61#diff-7ec4400f9507e8a5af187feecc073d1bR86
There is no `.yaml.default` for logs-only config at the moment.

Reply:
thanks for clarifying, I see there's a test covering this too 👍 

To me the opposite behavior (logs-only `.yaml` configs overriding `.yaml.default` configs) sounds more consistent in the long run, but we can re-discuss later if needed, I'm fine with keeping the current behavior for now at least.